### PR TITLE
Create `useTrackEvents` hook

### PR DESCRIPTION
**Pull request by xpaczka**
Ref #654 

Transform `TrackEvents` component into `useTrackEvents` hook

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import web3Onboard from "shared/utils/web3Onboard"
 // no-extraneous-dependencies.
 // eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (ioay) [inline src/index.tsx:8]: Why we import posthog-js module as dev dependency and we use every time: ` eslint-disable-next-line import/no-extraneous-dependencies` ?

File: src/shared/hooks/helpers.ts [hunk 2]
```diff
@@ -233,3 +236,14 @@ export function useMobileScreen() {
...
 }
+
+export function useTrackEvents() {
+  const location = useLocation()
+  const posthog = usePostHog()
+  const isMobile = useMobileScreen()
+
+  useEffect(() => {
+    posthog?.capture("$pageview", { url: location.pathname, isMobile })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (ioay) [inline src/shared/hooks/helpers.ts:247]: Do we need this rule here?

Comment (jagodarybacka) [reply src/shared/hooks/helpers.ts:247]: we can add `useEffectOnce` hook (I think we have it in the extension repo)

Comment (xpaczka) [reply src/shared/hooks/helpers.ts:247]: On second we should add the dependencies to array

File: src/shared/hooks/helpers.ts [hunk 3]
```diff
@@ -233,3 +235,13 @@ export function useMobileScreen() {
...
 }
+
+export function useTrackEvents() {
+  const location = useLocation()
+  const posthog = usePostHog()
+  const isMobile = useMobileScreen()
+
+  useEffect(() => {
+    posthog?.capture("$pageview", { url: location.pathname, isMobile })
+  }, [posthog, location, isMobile])
```
Comment (jagodarybacka) [inline src/shared/hooks/helpers.ts:246]: we should run if only once when posthog is ready to capture the event

File: package.json [hunk 4]
```diff
@@ -38,6 +38,7 @@
     "os-browserify": "^0.3.0",
     "patch-package": "^8.0.0",
     "path-browserify": "^1.0.1",
+    "posthog-js": "^1.88.2",
```
Comment (jagodarybacka) [inline package.json:41]: @Shadowfiend any reason why it was in dev dependencies?

**Comment by jagodarybacka** [comment]
I think pageview event props are not going through to posthog (or I don't know where to look) and it seems like we already have info what browser env was used so maybe we don't need it?

**Review by jagodarybacka — APPROVED** [review]

**Pull request merged**